vw/src/main/scala/com/microsoft/azure/synapse/ml/vw/VowpalWabbitPrediction.scala

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.vw
+
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.types.{ArrayType, FloatType, IntegerType, StructField, StructType}
+import org.vowpalwabbit.spark.VowpalWabbitNative
+import org.vowpalwabbit.spark.prediction.ScalarPrediction
+import vowpalWabbit.responses.{ActionProbs, ActionScores, DecisionScores, Multilabels, PDF, PDFValue}
+
+/**
+  * Provide schemas and accessor functions for almost all VW prediction types.
+  */
+object VowpalWabbitPrediction {
+  val SchemaMap = Map(
+    "prediction_type_t::scalars" -> Seq(StructField("predictions", ArrayType(FloatType), nullable = false)),
+    "prediction_type_t::multiclass" -> Seq(StructField("prediction", IntegerType, nullable = false)),
+    "prediction_type_t::prob" -> Seq(StructField("prediction", FloatType, nullable = false)),
+    "prediction_type_t::multilabels" -> Seq(StructField("prediction", ArrayType(IntegerType), nullable = false)),
+    "prediction_type_t::scalar" -> Seq(
+      StructField("prediction", FloatType, nullable = false),
+      StructField("confidence", FloatType, nullable = false)),
+    "prediction_type_t::action_scores" -> Seq(
+      StructField("predictions", ArrayType(StructType(Seq(
+        StructField("action", IntegerType, nullable = false),
+        StructField("score", FloatType, nullable = false)
+      ))))
+    ),
+    "prediction_type_t::action_probs" -> Seq(
+      StructField("predictions", ArrayType(StructType(Seq(
+        StructField("action", IntegerType, nullable = false),
+        StructField("probability", FloatType, nullable = false)
+      ))))
+    ),
+    "prediction_type_t::decision_probs" -> Seq(
+      StructField("predictions",
+        ArrayType(
+          ArrayType(
+            StructType(Seq(
+              StructField("action", IntegerType, nullable = false),
+              StructField("score", FloatType, nullable = false)
+            )))))),
+    "prediction_type_t::action_pdf_value" -> Seq(
+      StructField("action", FloatType, nullable = false),
+      StructField("pdf", FloatType, nullable = false)
+    ),
+    "prediction_type_t::pdf" -> Seq(
+      StructField("segments", ArrayType(
+        StructType(Seq(
+          StructField("left", FloatType, nullable = false),
+          StructField("right", FloatType, nullable = false),
+          StructField("pdfValue", FloatType, nullable = false)
+        )))))
+    )
```

Comment:
nit: Consider making a VW Schmea file anusing Case Classes + the SparkBindings trait like we have done for cognitive service schemas. Can make this much more readable

Reply:
done